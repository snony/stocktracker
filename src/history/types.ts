export interface HistoryData {
  readonly [key: string]: string | number
  readonly date: string
  readonly price: number
}

export interface HistoryState {
  history: HistoryData[]
  dateFilter: string
  priceFilter: string
}

export interface HistoryChartProps {
  readonly history: HistoryData[]
}

export interface FilterButtonProps {
  readonly type: string
  readonly value: string
  readonly selected: boolean
  readonly onClick: (symbol: string, value: string) => void
}
