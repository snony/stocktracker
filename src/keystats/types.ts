import { extractKeyStatsData } from '../api'

export type KeyStats = ReturnType<typeof extractKeyStatsData>
