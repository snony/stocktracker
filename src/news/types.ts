export enum FetchStatus {
  PENDING,
  SUCCESS,
  FAIL
}
export interface News {
  readonly url: string
  readonly headline: string
  readonly datetime: string
  readonly source: string
}

export interface NewsItemsProps {
  readonly fetchStatus: FetchStatus
  readonly newsItems: News[]
}
export interface NewsProps {
  readonly newsData: News
}
