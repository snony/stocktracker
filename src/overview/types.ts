export interface OverviewState {
  companyName: string
  symbol: string
  website: string
  description: string
}
