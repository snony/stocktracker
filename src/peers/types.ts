import { PeersState } from './peersReducer'

type Peer = string
export type Peers = Peer[]
export { PeersState }
