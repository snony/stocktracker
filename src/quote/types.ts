import { QUOTE } from 'socket/'
export interface QuoteProps {
    quote: QUOTE
    unsubscribe: () => void
}