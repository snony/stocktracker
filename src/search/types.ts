export type Company = {name: string, symbol:string};
export type ClickResult = (company:Company) => void;

export type Companies = Array<Company>

export interface SearchProps{
    readonly getInfo: ClickResult;
    readonly companySymbols: Companies; 
}

export interface SearchState{
    readonly searchValue:string;
    readonly suggestedCompanies: Companies;
}