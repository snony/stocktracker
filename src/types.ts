import api from './api'
import rootReducer from './reducer'
export type API = typeof api
export type GlobalState = ReturnType<typeof rootReducer>