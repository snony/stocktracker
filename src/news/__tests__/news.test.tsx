import React from 'react'
import renderer from 'react-test-renderer'

import NewsItems, { News } from '../news'
import { NewsItemsProps, NewsProps } from '../types'

describe('NewsItems Component', () => {
  it('renders correctly with empty newsItems passed to it', () => {
    const newsItems: NewsItemsProps = { newsItems: [] }
    const tree = renderer.create(<NewsItems {...newsItems} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with newsItem of length one passed to it', () => {
    const newsData: NewsProps = {
      newsData: {
        url: 'www.apple.com/news/apple_strategy',
        headline: 'Apple strategy is to win the electronic world war',
        datetime: '5/10/2018',
        source: 'The New York Post'
      }
    }
    const newsItems: NewsItemsProps = { newsItems: [newsData.newsData] }
    const tree = renderer.create(<NewsItems {...newsItems} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('News Component', () => {
  it('renders correctly the newsData', () => {
    const newsData: NewsProps = {
      newsData: {
        url: 'www.apple.com/news/apple_strategy',
        headline: 'Apple strategy is to win the electronic world war',
        datetime: '5/10/2018',
        source: 'The New York Post'
      }
    }
    const tree = renderer.create(<News {...newsData} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
