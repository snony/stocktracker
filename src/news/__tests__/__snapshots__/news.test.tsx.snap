// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News Component renders correctly the newsData 1`] = `
.emotion-1 {
  color: white;
  display: block;
  margin-bottom: 0.3rem;
  font-size: 0.9rem;
}

.emotion-0 {
  color: white;
  display: block;
  margin-bottom: 0.3rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div>
  <a
    href="www.apple.com/news/1"
  >
    <span
      className="emotion-0"
    >
      <span
        width={0}
      >
        <span />
        <span>
          Not Fake News
        </span>
        <span
          style={
            Object {
              "left": 0,
              "position": "fixed",
              "top": 0,
              "visibility": "hidden",
            }
          }
        >
          …
        </span>
      </span>
    </span>
  </a>
  <span
    className="emotion-1"
  >
    20/10/2018
     - 
    Seeking beta
  </span>
</div>
`;

exports[`NewsItems Component renders correctly news pending 1`] = `
.emotion-0 {
  color: white;
  display: block;
  margin-bottom: 0.3rem;
  font-size: 0.9rem;
}

<span
  className="emotion-0"
>
  ...Loading...
</span>
`;

exports[`NewsItems Component renders correctly on fetch status of fail 1`] = `
.emotion-0 {
  color: white;
  display: block;
  margin-bottom: 0.3rem;
  font-size: 0.9rem;
}

<span
  className="emotion-0"
>
  Cannot load data, check internet connection
</span>
`;

exports[`NewsItems Component renders correctly on fetch status of success and display newsItems 1`] = `
.emotion-1 {
  color: white;
  display: block;
  margin-bottom: 0.3rem;
  font-size: 0.9rem;
}

.emotion-10 {
  display: grid;
  grid-gap: 1.2rem;
  grid-template-columns: 1;
}

.emotion-0 {
  color: white;
  display: block;
  margin-bottom: 0.3rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
  className="emotion-10"
>
  <div>
    <a
      href="www.apple.com/news/1"
    >
      <span
        className="emotion-0"
      >
        <span
          width={0}
        >
          <span />
          <span>
            Not Fake News
          </span>
          <span
            style={
              Object {
                "left": 0,
                "position": "fixed",
                "top": 0,
                "visibility": "hidden",
              }
            }
          >
            …
          </span>
        </span>
      </span>
    </a>
    <span
      className="emotion-1"
    >
      20/10/2018
       - 
      Seeking beta
    </span>
  </div>
  <div>
    <a
      href="www.apple.com/news/2"
    >
      <span
        className="emotion-0"
      >
        <span
          width={0}
        >
          <span />
          <span>
            All eyes on treasury
          </span>
          <span
            style={
              Object {
                "left": 0,
                "position": "fixed",
                "top": 0,
                "visibility": "hidden",
              }
            }
          >
            …
          </span>
        </span>
      </span>
    </a>
    <span
      className="emotion-1"
    >
      20/10/2018
       - 
      SeekingAlpha
    </span>
  </div>
  <div>
    <a
      href="www.apple.com/news/3"
    >
      <span
        className="emotion-0"
      >
        <span
          width={0}
        >
          <span />
          <span>
            Apple won the fight against Apple
          </span>
          <span
            style={
              Object {
                "left": 0,
                "position": "fixed",
                "top": 0,
                "visibility": "hidden",
              }
            }
          >
            …
          </span>
        </span>
      </span>
    </a>
    <span
      className="emotion-1"
    >
      20/10/2018
       - 
      Seeking beta
    </span>
  </div>
  <div>
    <a
      href="www.apple.com/news/4"
    >
      <span
        className="emotion-0"
      >
        <span
          width={0}
        >
          <span />
          <span>
            Apple won the fight against Apple
          </span>
          <span
            style={
              Object {
                "left": 0,
                "position": "fixed",
                "top": 0,
                "visibility": "hidden",
              }
            }
          >
            …
          </span>
        </span>
      </span>
    </a>
    <span
      className="emotion-1"
    >
      20/10/2018
       - 
      Seeking beta
    </span>
  </div>
  <div>
    <a
      href="www.apple.com/news/5"
    >
      <span
        className="emotion-0"
      >
        <span
          width={0}
        >
          <span />
          <span>
            Apple won the fight against Apple
          </span>
          <span
            style={
              Object {
                "left": 0,
                "position": "fixed",
                "top": 0,
                "visibility": "hidden",
              }
            }
          >
            …
          </span>
        </span>
      </span>
    </a>
    <span
      className="emotion-1"
    >
      20/10/2018
       - 
      Seeking beta
    </span>
  </div>
</div>
`;
